<!DOCTYPE html>
<!--[if lt IE 7]>      <html class="no-js lt-ie9 lt-ie8 lt-ie7"> <![endif]-->
<!--[if IE 7]>         <html class="no-js lt-ie9 lt-ie8"> <![endif]-->
<!--[if IE 8]>         <html class="no-js lt-ie9"> <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js"> <!--<![endif]-->
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <title>BHSC: links</title>
        <meta name="description" content="">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <!-- Place favicon.ico and apple-touch-icon.png in the root directory -->
        <link rel="stylesheet" href="css/normalize.css">
        <link rel="stylesheet" href="css/main.css">
        <link rel="stylesheet" href="css/bhsc.css">
        <script src="js/vendor/modernizr-2.6.2.min.js"></script>
    </head>
    <body>
        <!--[if lt IE 7]>
            <p class="browsehappy">You are using an <strong>outdated</strong> browser. Please <a href="http://browsehappy.com/">upgrade your browser</a> to improve your experience.</p>
        <![endif]-->

        <!-- Add your site or application content here -->
        <div id="header"><h1>Bar Harbor Short Course on Medical and Experimental Mammalian Genetics</h1></div>
        <div id="content">
            <div id="sidebar"><h4>Pages</h4>
                <ul id="nav">
                    <li><a href="../index.html">Home</a></li>
                    <li><a href="../2014_07_21">Day 1</a></li>
                    <li><a href="../2014_07_23">Day 2</a></li>
                    <li><a href="../2014_07_24">Day 3</a></li>
                    <li><a href="../2014_07_25">Day 4</a></li>
                    <li><a href="../2014_07_29">Day 5</a></li>
                    <li><a href="refs.html">References</a></li>
                    <li><a href="formats.html">File Formats</a></li>
                    <li><a href="links.html">Resources</a></li>
                </ul></div>
            <div id="main"</div>
                <h1>Bioinformatic Resources</h1>
                <p>Below are a list of common databases and tools. This list is not exhaustive and may not be up to date. Listing of tools on this page is not an endorsement of the tool. </p>
                <fieldset class="group" id="db">
                <legend>Databases and genome viewers</legend>
                <p class="subtitle">A list of useful databases and online data viewers</p>
                <p><span class="strong">NCBI</span> (<a href="http://www.ncbi.nlm.nih.gov/" title="NCBI">http://www.ncbi.nlm.nih.gov/</a>): US sequence archive and biomedical data resource.</p>
                <p class="indent"><span class="strong">Education</span> (<a href="http://www.ncbi.nlm.nih.gov/education">http://www.ncbi.nlm.nih.gov/education</a>): information and documentation on using NCBI resources.</p>
                <p class="indent"><span class="strong">Gene</span> (<a href="http://www.ncbi.nlm.nih.gov/gene">http://www.ncbi.nlm.nih.gov/gene</a>): A database that aggregates information about genes.</p>
                <p class="indent"><span class="strong">GEO</span> (<a href="http://www.ncbi.nlm.nih.gov/geo">http://www.ncbi.nlm.nih.gov/geo</a>): A database of functional genomics data.</p>
                <p class="indent"><span class="strong">Epigenomics</span> (<a href="http://www.ncbi.nlm.nih.gov/epigenomics">http://www.ncbi.nlm.nih.gov/epigenomics</a>): A collection of genomic maps of epigenetic data.</p>
                <p class="indent"><span class="strong">dbSNP</span> (<a href="http://www.ncbi.nlm.nih.gov/snp">http://www.ncbi.nlm.nih.gov/snp</a>): A database of small scale genome variation.</p>
                <p class="indent"><span class="strong">dbVar</span> (<a href="http://www.ncbi.nlm.nih.gov/dbvar">http://www.ncbi.nlm.nih.gov/dbvar</a>): A database of large scale genomic structural variation.</p>
                <p class="indent"><span class="strong">Genome</span> (<a href="http://www.ncbi.nlm.nih.gov/genome">http://www.ncbi.nlm.nih.gov/genome</a>): access to genome information</p>
                <p class="indent"><span class="strong">CloneDB</span> (<a href="http://www.ncbi.nlm.nih.gov/clone">http://www.ncbi.nlm.nih.gov/clone</a>): A catalog of genomic and cell line based clones.</p>
                <p class="indent"><span class="strong">Genome Workbench</span> (<a href="http://www.ncbi.nlm.nih.gov/projects/gbench">http://www.ncbi.nlm.nih.gov/projects/gbench</a>): A client side genome viewer and analyis tool.</p>
                <p class="indent"><span class="strong">SRA</span> (<a href="http://www.ncbi.nlm.nih.gov/sra">http://www.ncbi.nlm.nih.gov/sra</a>): An archive of next generation sequences.</p>
                <p class="indent"><span class="strong">Pubmed</span> (<a href="http://www.ncbi.nlm.nih.gov/pubmed">http://www.ncbi.nlm.nih.gov/pubmed</a>): Literature resource.</p>
                <p><span class="strong">Ensembl</span> (<a href="http://www.ensembl.org/index.html">http://www.ensembl.org/index.html</a>): genome databases for vertebrates and other eukaryotic species.</p>
                <p><span class="strong">UCSC</span> (<a href="http://genome.ucsc.edu/">http://genome.ucsc.edu/</a>): browser for genome data of selected species.</p>
                <p><span class="strong">Mouse Genome Informatics</span> (<a href="http://www.informatics.jax.org/">http://www.informatics.jax.org/</a>): Model organism database for mouse</p>
                <p><span class="strong">International Gene Trap Consortium</span> (<a href="http://www.genetrap.org/">http://www.genetrap.org/</a>): Database of trapped genes. </p>
                <p><span class="strong">International Mouse Knockout database</span> (<a href="http://www.knockoutmouse.org/">http://www.knockoutmouse.org/</a>): Database of knocked out genes (includes gene traps).</p>
                <p><span class="strong">Rat Genome Database</span> (<a href="http://rgd.mcw.edu/">http://rgd.mcw.edu/</a>): Model organism database for rat.</p>
                <p><span class="strong">Genome Reference Consortium</span> (<a href="http://genomereference.org/">http://genomereference.org</a>): Curation of human, mouse and zebrafish reference genome assemblies.</p>
                <p><span class="strong">Map Viewer:</span> (<a href="http://www.ncbi.nlm.nih.gov/mapview">http://www.ncbi.nlm.nih.gov/mapview</a>): Genome tool for viewing multiple coordinate systems (assemblies, maps, etc) simultaneously.</p>
                <p><span class="strong">Sequence Viewer</span> (<a href="http://www.ncbi.nlm.nih.gov/projects/sviewer">http://www.ncbi.nlm.nih.gov/projects/sviewer</a>): A graphical view you will see on many NCBI pages. This tools allows for viewing of annotations on sequence data and provides access to some analysis tools.</p>
                <p><span class="strong">Genome Workbench</span> (<a href="http://www.ncbi.nlm.nih.gov/tools/gbench">http://www.ncbi.nlm.nih.gov/tools/gbench</a>): A client side tool for viewing and analyzing sequence data.</p>
                <p><span class="strong">Integrated Genomics Viewer</span> (<a href="http://www.broadinstitute.org/software/igv/">http://www.broadinstitute.org/software/igv/</a>): A client side tool for viewing large scale genome data.</p>
                <p><span class="strong">The 1000 Genomes Browser</span> (<a href="http://www.ncbi.nlm.nih.gov/variation/tools/1000genomes">http://www.ncbi.nlm.nih.gov/variation/tools/1000genomes</a></p>
                </fieldset>
                <fieldset class="group" id="analysis">
                <legend>Analysis Resources</legend>
                <p><span class="strong">Biostar</span> (<a href="http://biostars.org/">http://biostars.org/</a>): An online resource to ask questions about bioinformatics.</p>
                <p><span class="strong">SeqAnswers</span> (<a href="http://seqanswers.com/">http://seqanswers.com/</a>): An online forum for asking questions about sequencing and sequence analysis.</p>
                <p><span class="strong">Galaxy</span> (<a href="http://main.g2.bx.psu.edu/">http://main.g2.bx.psu.edu/</a>): a resource that allows you to do some bioinformatic analysis without having to learn to program.</p>
                <p><span class="strong">List of public Galaxy Installations</span> (<a href="http://wiki.galaxyproject.org/PublicGalaxyServers">http://wiki.galaxyproject.org/PublicGalaxyServers</a> A list of other public Galaxy installations.</p>
                <p><span class="strong">Gene Pattern</span> (<a href="http://www.broadinstitute.org/cancer/software/genepattern/">http://www.broadinstitute.org/cancer/software/genepattern/</a>): A genomic analysis platform from the Broad Institute.</p>
                <p><span class="strong">GenomeSpace</span> (<a href="http://www.genomespace.org/">http://www.genomespace.org</a>): A resource to allow scientists without programming skills to access genome analysis tools.</p>
                <p><span class="strong">GenPlay</span> (<a href="http://genplay.einstein.yu.edu/wiki/index.php/Main_Page">http://genplay.einstein.yu.edu/wiki/index.php/Main_Page</a>): A genomic data analyzer for the Albert Einstein College of Medicine.</p>
                <p><span class="strong">NCBI Remap</span> (<a href="http://www.ncbi.nlm.nih.gov/genome/tools/remap">http://www.ncbi.nlm.nih.gov/genome/tools/remap</a>): A tool for projecting features from one assembly to another or from an assembly to a RefSeqGene. </p>
                <p><span class="strong">liftOver</span> (<a href="http://genome.ucsc.edu/cgi-bin/hgLiftOver">http://genome.ucsc.edu/cgi-bin/hgLiftOver</a>): A tool for projecting features from one assembly to another. </p>
                </fieldset>
                <fieldset class="group" id="align">
                <legend>Alignment software</legend>
                <p class="subtitle">This list of software contains links for well established aligners for larger sequences (typically those in GenBank/EMBL/DDBJ). There is rapid software development for Next Generation Sequencing applications, so rather than list all options, I've linked to two sites that are mainting lists of these applications.</p>
                <p><span class="strong">BLAST</span> (<a href="http://blast.ncbi.nlm.nih.gov/Blast.cgi">http://blast.ncbi.nlm.nih.gov/Blast.cgi</a>): a suite of alignment tools designed to align nucleotides and protein.        </p>
                <p><span class="strong">SPLIGN</span> (<a href="http://www.ncbi.nlm.nih.gov/sutils/splign/splign.cgi">http://www.ncbi.nlm.nih.gov/sutils/splign/splign.cgi</a>): a tool specifically designed to aligned spliced sequences to genomic DNA.        </p>
                <p><span class="strong">BLAT</span> (<a href="http://genome.ucsc.edu/cgi-bin/hgBlat">http://genome.ucsc.edu/cgi-bin/hgBlat</a>): find regions of similiarity between closely related biological sequences (nucleotide or protein).        </p>
                <p><span class="strong">SeqAnswers</span> list of NGS software (    <a href="http://seqanswers.com/forums/showthread.php?t=43">http://seqanswers.com/forums/showthread.php?t=43</a>)      </p>
                <p><span class="strong">Wikipedia</span> list of alignment software (<a href="http://en.wikipedia.org/wiki/List_of_sequence_alignment_software">http://en.wikipedia.org/wiki/List_of_sequence_alignment_software</a>): This pages is not limited to NGS aligners.</p>
                <p><span class="strong">HTS Mappers list</span> list of HTS mappers with feature comparisons. (<a href="http://wwwdev.ebi.ac.uk/fg/hts_mappers/">http://wwwdev.ebi.ac.uk/fg/hts_mappers</a></p>
                </fieldset>
                <fieldset class="group" id="software">
                <legend>Software toolkits</legend>
                <p class="subtitle">Software libraries for writing your own processes</p>
                <p><span class="strong">SRA Toolkit</span> (<a href="http://www.ncbi.nlm.nih.gov/books/NBK47540">http://www.ncbi.nlm.nih.gov/books/NBK47540</a>/): software for manipulating data stored in the SRA.</p>
                <p><span class="strong">NCBI C++ toolkit</span> (<a href="http://www.ncbi.nlm.nih.gov/books/NBK7160/">http://www.ncbi.nlm.nih.gov/books/NBK7160/</a>): library code for managing sequencing data. </p>
                <p><span class="strong">E-utilities</span> (<a href="http://eutils.ncbi.nlm.nih.gov/">http://eutils.ncbi.nlm.nih.gov/</a>): modules for using Perl to query the Entrez search system.</p>
                <p><span class="strong">GATK</span> (<a href="http://www.broadinstitute.org/gsa/wiki/index.php/The_Genome_Analysis_Toolkit">http://www.broadinstitute.org/gsa/wiki/index.php/The_Genome_Analysis_Toolkit</a>): toolkit for manipulating next generation sequencing data. </p>
                <p><span class="strong">SAMtools</span> (<a href="http://sourceforge.net/projects/samtools/">http://sourceforge.net/projects/samtools/</a>): tools for manipulating SAM/BAM files.</p>
                <p><span class="strong">VCFtools</span> (<a href="http://vcftools.sourceforge.net/">http://vcftools.sourceforge.net/</a>): tools for manipulating VCF.</p>
                </fieldset>
                <fieldset class="group" id="code">
                <legend>Learn to code</legend>
                <p class="subtitle">Resources for learning to write your own code</p>
                <p><span class="strong">Software Carpentry</span> (<a href="http://software-carpentry.org/">http://software-carpentry.org</a>): Introduction to many aspects of producing quality software.</p>
                <p><span class="strong">Code Academy</span> (<a href="http://www.codeacademy.com/">http://www.codeacademy.com</a>): Interactive web site that teaches you to code by coding.</p>
                <p><span class="strong">Learn Python the Hard way</span>(<a href="http://www.ncbi.nlm.nih.gov/staff/church/GenomeAnalysis/learnpythonthehardway.org">http://learnpythonthehardway.org</a>): Site for learning Python.</p>
                <p><span class="strong">List of courses and Tutorials</span>(<a href="http://stephenturner.us/p/edu">http://stephenturner.us/p/edu</a>): A web page maintained by Stephen Turner linking to in person classes and on-line material for NGS analyis.</p>
                <p><span class="strong">Scientific computing modules for Python</span>(<a href="http://www.lowindata.com/2013/installing-scientific-python-on-mac-os-x/">http://www.lowindata.com/2013/installing-scientific-python-on-mac-os-x/</a>): A list of useful modules for scientific computing using Python and how to install them (MacOX).</p>
                </fieldset>
                <fieldset class="group" id="tutorials">
                <legend>On line tutorials</legend>
                <p class="subtitle">On-line tutorials for specific tools and applications</p>
                <p><span class="strong">UC Davis RNA-Seq tutorial</span> (<a href="http://training.bioinformatics.ucdavis.edu/docs/2012/05/RNA/index.html">http://training.bioinformatics.ucdavis.edu/docs/2012/05/RNA/index.html</a>): An in depth guide to performing RNA-Seq analysis.</p>
                </fieldset>   
            </div>
        </div>
        <div class="cleaner"> </div>
        <div id="footer">Page last updated: <span id="date"></span></div>
        <script src="//ajax.googleapis.com/ajax/libs/jquery/1.10.2/jquery.min.js"></script>
        <script>window.jQuery || document.write('<script src="js/vendor/jquery-1.10.2.min.js"><\/script>')</script>
        <script src="js/plugins.js"></script>
        <script src="js/main.js"></script>
        <script src="js/jquery-ui-1.11.0/jquery-ui.min.js"></script>
        <script type="text/javascript">
            var d=new Date(document.lastModified)
            d1=jQuery.datepicker.formatDate('dd-mm-yy', d)
            jQuery("#date").append(d1)
        </script>
    </body>
</html>
